packages/core/src/entity/Collection.ts

```diff
@@ -335,7 +342,7 @@ export class Collection<T extends object, O extends object = object> extends Arr
     }
 
     const populate = Array.isArray(options.populate)
-      ? options.populate.map(f => `${this.property.name}.${f}`)
+      ? [...(options.populate.includes('*') ? Object.keys(this.property.targetMeta!.properties) : options.populate)].map(f => `${this.property.name}.${f}`)
```

Comment:
This looks wrong, the star needs to work recursively and you are only expanding one level here.

The idea was that we pass down the star and it's gonna be handled in the EntityLoader. So it should be rather this:

```suggestion
      ? options.populate.map(f => f === '*' ? f : `${this.property.name}.${f}`)
```

Reply:
I've tried your suggestion but `await Promise.all(colsA.map(col => col.loadItems({ populate: [ '*' ] })));` times out.

Reply:
Yes, because there is some bug in there, I saw this in another test just a few days ago. But it's the correct way of doing this.

Reply:
Rebased against https://github.com/mikro-orm/mikro-orm/pull/5001